### PR TITLE
Firecert 2080

### PR DESCRIPTION
**Pull request by Roebot926**

### CODE REVIEW

File: cypress/support/cypress-commands/assertion.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ Cypress.Commands.add(
                 cy.log(
                   `Actual error code ${apiErrorResponse.code} expected to be present in list of expected error codes`
                 ).then(() => {
-                  assert.include(
+                  fireLog.include(
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:104]: The cy log above this, that should have been internally handled by firelog, correct?

File: cypress/support/cypress-commands/assertion.js [hunk 2]
```diff
@@ -113,20 +113,20 @@ Cypress.Commands.add(
                 cy.log(
                   `Actual error message ${apiErrorResponse.message} expected to be present in list of expected error messages`
                 ).then(() => {
-                  assert.equal(checkErrorMessage, true, 'Error Message Validation: ');
+                  fireLog.equal(checkErrorMessage, true, 'Error Message Validation: ');
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:116]: Similar comment as above

File: cypress/support/cypress-commands/assertion.js [hunk 3]
```diff
@@ -113,20 +113,20 @@ Cypress.Commands.add(
...
                 });
               }
             );
           });
         } else {
           cy.log('Unable to find data for Error validation').then(() => {
-            assert(false, 'Unable to find data for Error validation');
+            fireLog(false, 'Unable to find data for Error validation');
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:123]: We don't need the cy.log

File: cypress/support/cypress-commands/assertion.js [hunk 4]
```diff
@@ -113,20 +113,20 @@ Cypress.Commands.add(
...
           });
         }
       });
     } catch (error) {
       cy.log('Failed to validate error: ', error).then(() => {
-        assert(false, 'Failed to validate error: ' + error);
+        fireLog(false, 'Failed to validate error: ' + error);
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:129]: Remove cy.log

File: cypress/support/cypress-commands/assertion.js [hunk 5]
```diff
@@ -158,16 +158,14 @@ Cypress.Commands.add(
...
-        assert.deepEqual(apiResponseContent, expected, pretext);
-      });
+      const pretext =
+        CONSTANTS.METHOD_CONTENT +
+        ' expected ' +
+        JSON.stringify(apiResponseContent) +
+        ' to be ' +
+        JSON.stringify(expected);
+      // Executing fireLog.deepEqual() after logging
+      fireLog.deepEqual(apiResponseContent, expected, pretext);
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:168]: Do we need pretext here? Doesn't firelog already create a message?
Can't this just be
`fireLog.deepEqual(apiResponseContent, expected, CONSTANTS.METHOD_CONTENT);`

Also remove the comment `// Executing fireLog.deepEqual() after logging`

File: cypress/support/cypress-commands/assertion.js [hunk 6]
```diff
@@ -471,7 +471,7 @@ Cypress.Commands.add(
         const pretext = 'Event Not Received : ';
 
         cy.log(pretext + ': expected ' + eventResponse + ' to be ' + content).then(() => {
-          assert.strictEqual(eventResponse, content, pretext);
+          fireLog.strictEqual(eventResponse, content, pretext);
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:474]: Do we need above cy log?

File: cypress/support/cypress-commands/assertion.js [hunk 7]
```diff
@@ -553,7 +553,7 @@ Cypress.Commands.add(
           const pretext = 'Event Not Received : ';
 
           cy.log(pretext + ': expected ' + eventResponse + ' to be ' + content);
-          assert.strictEqual(eventResponse, content, pretext);
+          fireLog.strictEqual(eventResponse, content, pretext);
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:556]: Remove `cy.log.then()` wherever applicable. The logging would've happened internally in firelog

Comment (parag-pv) [reply cypress/support/cypress-commands/assertion.js:556]: Fix this in all occurrences

File: cypress/support/cypress-commands/commands.js [hunk 8]
```diff
@@ -433,7 +433,7 @@ Cypress.Commands.add('getBeforeOperationObject', () => {
         }
       });
     } else {
-      assert(
+      fireLog(
         false,
         'Before operation object is not in proper array format, recheck the before objects in fixture/external/moduleReqId - getBeforeOperationObject'
```
Comment (parag-pv) [inline cypress/support/cypress-commands/commands.js:438]: Can we move this string to constants?

File: cypress/support/cypress-commands/commands.js [hunk 9]
```diff
@@ -845,7 +845,7 @@ Cypress.Commands.add(
   'combineValidationJson',
   (fireboltCallsValidationPathFromFCS, fireboltCallsValidationPathFromConfigModule) => {
     cy.mergeJsonfiles(fireboltCallsValidationPathFromFCS).then((fCSFixturesValidationjson) => {
-      assert.isNotNull(
+      fireLog.isNotNull(
         fCSFixturesValidationjson,
         'Expected JSON data should be defined in fixtures/objects/validationObjects/'
```
Comment (parag-pv) [inline cypress/support/cypress-commands/commands.js:850]: Can we move this string to constants?

File: cypress/support/cypress-commands/assertion.js (hunk 5 above)
Comment (Roebot926) [reply cypress/support/cypress-commands/assertion.js:168]: Pretext can be removed, but removing it changes the log from 

[5/10/2024 11:23:57 AM]["Method Content validation expected \"1395767607303387913\" to be \"1395767607303387913\""]

to 

[5/10/2024 11:25:36 AM]["1395767607303387913"]

File: cypress/support/cypress-commands/assertion.js (hunk 7 above)
Comment (Roebot926) [reply cypress/support/cypress-commands/assertion.js:542]: Might want to double down and check again, but I am pretty sure I got all of them. I looked over it a few times, hopefully I did not miss any.

File: cypress/support/cypress-commands/assertion.js [hunk 10]
```diff
@@ -422,24 +408,26 @@ function loggingValidationCheckResult(validationCheck) {
...
 
-  // Printing the status of all checks in the report.
-  cy.get(validationCheck)
-    .each((logging) => {
-      cy.log(
-        `${logging.validationPoint}: ${logging.validationStatus}. ${logging.message}`,
-        'loggingValidationCheckResult'
+  // Constructing the logString to pass to fireLog
+  // Assume the checks. If anything is marked other than skipped or pass, then fail the testcase.
+  validationCheck.forEach((assertion) => {
```
Comment (parag-pv) [inline cypress/support/cypress-commands/assertion.js:413]: I foresee a problem here. The reason for the loops is we need logs for all assertions even if one of the assertion fails. I suspect that the use of `firelog` here might break that.

To verify this, make the schema validation fail or run any scenario where schema validation is failing in an existing report. You should see the logs for all 4 validations. If not, changes in this block has to be reverted. 
We can create a new ticket to optimize this flow separately.

**Review by parag-pv — DISMISSED** [review]

**Review by parag-pv — APPROVED** [review]

**Review by neeradanelxsi — APPROVED** [review]

**Review by parag-pv — APPROVED** [review]

**Pull request merged**